app/src/test/java/org/mozilla/fenix/utils/BrowsersCacheTest.kt

```diff
@@ -0,0 +1,146 @@
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.TestApplication
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+@Config(application = TestApplication::class)
+class BrowsersCacheTest {
+
+    // NB: There is always one more browser than pretendBrowsersAreInstalled installs because
+    // the application we are testing is recognized as a browser itself!
+
+    @Test
+    fun testInstallNewBrowsersWithoutReset() {
+        pretendBrowsersAreInstalled(
+            browsers = listOf(
+                Browsers.KnownBrowser.FIREFOX_NIGHTLY.packageName,
+                Browsers.KnownBrowser.REFERENCE_BROWSER.packageName
+            )
+        )
+
+        var browsers = BrowsersCache.all(testContext)
+        assertEquals(browsers.installedBrowsers.size, 3)
+
+        pretendBrowsersAreInstalled(
+            browsers = listOf(
+                Browsers.KnownBrowser.FIREFOX_NIGHTLY.packageName,
+                Browsers.KnownBrowser.FIREFOX.packageName,
+                Browsers.KnownBrowser.CHROME.packageName,
+                Browsers.KnownBrowser.SAMSUNG_INTERNET.packageName,
+                Browsers.KnownBrowser.DUCKDUCKGO.packageName,
+                Browsers.KnownBrowser.REFERENCE_BROWSER.packageName
+            )
+        )
+        browsers = BrowsersCache.all(testContext)
+        assertEquals(browsers.installedBrowsers.size, 3)
+    }
+
+    @Test
+    fun testInstallNewBrowsersWithReset() {
+        pretendBrowsersAreInstalled(
+            browsers = listOf(
+                Browsers.KnownBrowser.FIREFOX_NIGHTLY.packageName,
+                Browsers.KnownBrowser.REFERENCE_BROWSER.packageName
+            )
+        )
+
+        var browsers = BrowsersCache.all(testContext)
+        assertEquals(browsers.installedBrowsers.size, 3)
+
+        pretendBrowsersAreInstalled(
+            browsers = listOf(
+                Browsers.KnownBrowser.FIREFOX_NIGHTLY.packageName,
+                Browsers.KnownBrowser.FIREFOX.packageName,
+                Browsers.KnownBrowser.CHROME.packageName,
+                Browsers.KnownBrowser.SAMSUNG_INTERNET.packageName,
+                Browsers.KnownBrowser.DUCKDUCKGO.packageName,
+                Browsers.KnownBrowser.REFERENCE_BROWSER.packageName
+            )
+        )
+
+        BrowsersCache.resetAll()
+
+        browsers = BrowsersCache.all(testContext)
+        assertEquals(browsers.installedBrowsers.size, 7)
+    }
+
+    @Test
+    fun testResetAll() {
+        BrowsersCache.all(testContext)
+
+        assertNotNull(BrowsersCache.cachedBrowsers)
+        assertNotNull(BrowsersCache.cachedContext)
+
+        BrowsersCache.resetAll()
+
+        assertNull(BrowsersCache.cachedBrowsers)
+        assertNull(BrowsersCache.cachedContext)
+    }
+
+    private fun pretendBrowsersAreInstalled(
+        browsers: List<String> = listOf(),
+        defaultBrowser: String? = null,
```

Comment:
nit: afaict, none of these arguments are used so we should remove them to simplify the code.

Reply:
That is true, but they are useful to have *if* we do need them. I took this from a different test and they are important there. I think that we should leave them because there may come a time where it is necessary to create contexts using them for the purposes of testing this cache.

Reply:
Can we move this to a shared helper function then? I don't see any precident for this in the Fenix codebase but I usually add shared test code files to a `org.mozilla.fenix.helpers` package.